guides/shipments/custom-shipping-calculators.md

```diff
@@ -50,15 +50,115 @@ Typically, a calculator uses the following order information:
 calculator collects enough information about an order to send to a carrier and
 get a rate quote back.
 
 [flat-rate-source]: https://github.com/solidusio/solidus/blob/master/core/app/models/spree/calculator/shipping/flat_rate.rb
 [base-calculator-source]: https://github.com/solidusio-contrib/solidus_active_shipping/blob/master/app/models/spree/calculator/shipping/active_shipping/base.rb
 
+### Inherit from the `Spree::ShippingCalculator` class
+
+Your custom shipping calculator should inherit from the existing
+`Spree::ShippingCalculator` class. We recommend following the same directory
+structure that Spree models do, so your new calculator would be created at:
+
+```shell
+/app/models/my_store/calculator/shipping/custom_shipping_calculator.rb
+```
+
+Then, follow the pattern of Spree's built-in calculators and inherit from
+`Spree::ShippingCalculator`:
+
+```ruby
+module MyStore
+  class Calculator::Shipping::CustomShippingCalculator < Spree::ShippingCalculator
+  end
+end
+```
+
+### Required methods
+
+Your custom shipping calculator requires at least two methods:
+
+1. A `self.description` method that provides a name for the custom calculator.
+2. A `compute_package(package)` that provides the return value for a package
+   being shipped.
+
+For example:
+
+```ruby
+module MyStore
+  class Calculator::Shipping::CustomShippingCalculator < Spree::ShippingCalculator
+    def self.description
+      "Custom Shipping Calculator"
+    end
+
+    def compute_package(package)
+      12.00
+    end
+  end
+end
+```
+
+<!-- TODO:
+  This example code gets developers some of the way there, but could be improved
+  to be more realistic. See the built-in shipping calculators for more realistic
+  `compute_package` definitions.
+-->
+
+### Register the new shipping calculator
+
+Once you have created the logic for the new shipping calculator, you need to
+register it so that administrators can create new shipping methods that use the
+custom calculator.
+
+For example, you can register it in your `/config/application.rb` initializer:
+
+```ruby
+# /config/application.rb
+module MyStore
+  class Application < Rails::Application
+    ...
+
+    initializer 'spree.register.calculators' do |app|
+      app.config.spree.calculators.shipping_methods << MyStore::Calculator::Shipping::CustomShippingCalculator
+    end
+  end
+end
+```
+
+## Calculator availability
+
+By default, shipping calculators are always available to be used by shipping
+methods. This is because the `available?` method on the base
+[`Spree::Calculator` class][calculator-class] returns `true` by default.
+
+You may want to make the calculator availability change depending on some aspect
+of the current order. To do this, you can override the `available?` method in
+your custom calculator:
+
+```ruby
+module MyStore
+  class Calculator::Shipping::CustomShippingCalculator < Spree::ShippingCalculator
+    ...
+
+    def available?(object)
```

Comment:
I would recommend users against defining something as `object`. What is the object being passed in here? Lets explain that.

Reply:
I'm trying to check that this is possible now. Would it make sense to change `object` to `order` for this example? Or is the whole example kind of unrealistic, maybe?

Reply:
Yup. That makes sense. The example is fine as is as a trivial example, but will be helpful for people wondering what `object` is.